policies/templates/gcp_sql_allowed_authorized_networks_v1.yaml

```diff
@@ -71,9 +71,14 @@ spec:
             	asset.asset_type == "sqladmin.googleapis.com/Instance"
             
             	allowed_authorized_networks = lib.get_default(params, "authorized_networks", [])
-            	configured_networks := {network |
-            		network = asset.resource.settings.ipConfiguration.authorizedNetworks[_].value
-            	}
+            	
+                # Check whether authorizedNetworks field exists, so that
+                # we can report violation when this field is not set
+                config_auth_networks = lib.get_default(asset.resource.data.settings.ipConfiguration, "authorizedNetworks", [{"value":"authorized network unspecified"}])
+
+                configured_networks := {network |
+                network = config_auth_networks[_].value
+                }
             
             	matched_networks := {network |
```

Comment:
Instead of using a comprehension here, it should be sufficient to cast `allowed_authorized_networks` to a set then compare the two sets (`forbidden_networks := configured_networks - allowed_authorized_networks`).

Reply:
`allowed_authorized_networks` is the full whitelist, which I think is not what we want to compare here. We want to compare `configured_networks` with the networks matched to `allowed_authorized_networks`. That's why the comparison has to be between `configured_networks` and `matched_networks`.

As for `configured_networks`, I think it has to be a comprehension because we want to extract each value from `config_auth_networks`, which itself is an array of key/value maps.